trl/trainer/sft_trainer.py

```diff
@@ -150,8 +152,41 @@ def torch_call(self, examples: list[Union[list[int], Any, dict[str, Any]]]) -> d
             )
             output["labels"][completion_mask == 0] = -100  # mask everything that is not in the completion
 
+        if self.sequence_parallel_size > 1:
+            # Get local (start, end) for sequence parallelism slicing
+            total_seq_len = output["input_ids"].shape[1]
+            slice_size = total_seq_len // self.local_world_size
+            start = self.local_rank * slice_size
+            end = start + slice_size
+
+            # Generate proper position IDs for this rank
+            if "position_ids" not in output:
```

Comment:
position ids have to be sharded more accurately

see https://github.com/OpenRLHF/OpenRLHF/pull/439/files#diff-ad125e450b6a7837d6a8a74551924607aa4e46e6250664eb74d3bab9a5e4f6e2R17